theseus/optimizer/manifold_gaussian.py

```diff
@@ -21,10 +21,9 @@ def __init__(
         name: Optional[str] = None,
     ):
         self._id = next(ManifoldGaussian._ids)
-        if name:
-            self.name = name
-        else:
+        if name is None:
             self.name = f"{self.__class__.__name__}__{self._id}"
+        self.name = name
```

Comment:
This line will override with `None`. You need the line above to be `name = ` instead of `self.name = `.